internal/backend/user.go

```diff
@@ -93,3 +99,20 @@ func (user *user) close(ctx context.Context) error {
 
 	return nil
 }
+
+func (user *user) deleteAllMessagesMarkedDeleted() error {
+	ctx := context.Background()
```

Comment:
Best not to "hide" internal uses of context. If an operation needs context, prefer passing it down from as high in the call chain as possible. In this case, I would suggest adding `ctx` to `backend.AddUser` and `newUser`. If someone using the backend wants to add a user, and doing so triggers some potentially expensive DB operation, they'll want to be able to stop the operation by cancelling a context.